src/scripts/themeLoader.js

```diff
@@ -16,6 +16,12 @@ define(['userSettings', 'skinManager', 'connectionManager', 'events'], function
                 context = 'serverdashboard';
             } else {
                 theme = userSettings.theme();
+                var sysTheme = userSettings.followSystemTheme();
+                if (true === sysTheme) {
+                    if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
```

Comment:
This could likely be replaced by a function in NativeShell, to give the native client the opportunity to override the detection, if the host's webview doesn't provide this property.

You could look at [this](https://github.com/jellyfin/jellyfin-web/blob/master/src/components/apphost.js#L310) to see how to handle simple cases (Basically check if NativeShell is defined and defines the function, then execute that, otherwise use the media query way).

So it would be something like:
```suggestion
                    if (appHost.useDarkTheme()) {
```
Which would then set the theme as you do here already.

Reply:
We could also imagine that instead of returning a boolean, the appHost function returns either "light" or "dark", which then gets passed to a switch to load the correct theme in.

Reply:
> You could look at [this](https://github.com/jellyfin/jellyfin-web/blob/master/src/components/apphost.js#L310) to see how to handle simple cases (Basically check if NativeShell is defined and defines the function, then execute that, otherwise use the media query way).

https://github.com/jellyfin/jellyfin-web/blob/0989b5222c66aa5a87a9273b2ae794f43e185a48/src/scripts/site.js#L206-L212

Reply:
Oh, yes, that one is much better as an example lol

Reply:
Thanks for those suggestions. I have moved `useDarkTheme` to appHost and added a feature called `darktheme` in appHost. If the browser supports `darktheme`, the settings are visible, if not, they are not visible. Hopefully that's everything. If there is anything else that needs changing let me know